substrate/tendermint/client/src/block_import.rs

```diff
@@ -0,0 +1,186 @@
+use std::{marker::PhantomData, sync::Arc, collections::HashMap};
+
+use async_trait::async_trait;
+
+use sp_api::BlockId;
+use sp_runtime::traits::{Header, Block};
+use sp_blockchain::{BlockStatus, HeaderBackend, Backend as BlockchainBackend};
+use sp_consensus::{Error, CacheKeyId, BlockOrigin, SelectChain};
+
+use sc_consensus::{BlockCheckParams, BlockImportParams, ImportResult, BlockImport, Verifier};
+
+use sc_client_api::{Backend, BlockBackend};
+
+use crate::{TendermintValidator, tendermint::TendermintImport};
+
+impl<T: TendermintValidator> TendermintImport<T> {
+  fn check_already_in_chain(&self, hash: <T::Block as Block>::Hash) -> bool {
+    let id = BlockId::Hash(hash);
+    // If it's in chain, with justifications, return it's already on chain
+    // If it's in chain, without justifications, continue the block import process to import its
+    // justifications
+    // This can be triggered if the validators add a block, without justifications, yet the p2p
+    // process then broadcasts it with its justifications
+    if (self.client.status(id).unwrap() == BlockStatus::InChain) &&
+      self.client.justifications(hash).unwrap().is_some()
+    {
+      return true;
+    }
+    false
```

Comment:
That'll raise a warning in clippy for being pointless. If we then do further things, this won't compile for the missing semicolon. If someone adds a semicolon without thinking about it, it'll raise a warning about unused value.

This is how Rust code is written and it will break linters while going against convention if we try to change that.

Reply:
... why didn't I just write the condition itself as the tail value? We don't need an if statement to return true if true and false if false. That's a change I'd like to make since I don't see this function changing in any notable way. Good with you?